build.sh

```diff
@@ -89,6 +89,10 @@ for task in dcind:1.0.0 oci-build:0.9.0; do
     --platform linux/arm64 \
     --build-arg ${_b}_task_version=${_v} \
     --tag $DOCKER_REGISTRY_BASE/concourse-${_t}-task:latest \
-    --tag $DOCKER_REGISTRY_BASE/concourse-${_t}-task:${_v} \
-    --push .)
+    --tag $DOCKER_REGISTRY_BASE/concourse-${_t}-task:${_v} .)
+
+  if [ "$SHOULD_PUSH" = "true" ]; then
+    docker push $DOCKER_REGISTRY_BASE/concourse-${_t}-task:latest
```

Comment:
docker push without a tag, pushes all the available tags in the current workflow.